source/pkg/reconciler/streamsource/config.go

```diff
@@ -101,21 +101,19 @@ func GetRedisConfig(configMap map[string]string) (*RedisConfig, error) {
 	return config, nil
 }
 
-// GetTLSConfig returns the details of the TLS certificate.
-func GetTLSConfig(configMap map[string]string) (*TLSConfig, error) {
-	if len(configMap) == 0 {
+// GetTLSSecret returns the details of the TLS certificate.
+func GetTLSSecret(secret map[string][]byte) (*TLSConfig, error) {
+	if len(secret) == 0 {
 		return nil, fmt.Errorf("missing configuration")
 	}
 
+	//Convert byte to a string
 	config := &TLSConfig{
-		TLSCertificate: "",
+		TLSCertificate: string(secret[tlsConfigKey]),
 	}
 
-	err := configmap.Parse(configMap,
-		configmap.AsString(tlsConfigKey, &config.TLSCertificate),
-	)
-	if err != nil {
-		return nil, err
+	if config == nil {
```

Comment:
`config` will never be nil here since you are setting TLSCertificate to be string() of a byte array. And string() of an empty byte array which is what is returned when a key doesn't exist in a map, is an empty string "".

So a better correct check would be 
```
if config.TLSCertificate == "" {
    return nil, fmt.Errorf("tls certificate missing from secret")
 }
 ```
 Do you agree?